### PR TITLE
Validate current subpass against pipeline subpass

### LINKED ISSUE #933: Layers does not check if proper pipeline is bound after vkCmdNextSubpass
Consider following scenario:

```
vkCmdBeginRenderpass()
vkCmdBindPipeline(A); 
vkCmdDraw();
vkCmdNextSubpass();
vkCmdDraw();   <- Wrong, because pipeline A subpass index is 0, while current subass index is 1
```

Validation layers should validate pipeline subpass index wrt subpass index in each drawcall - not only when binding the pipeline.

### PR DESCRIPTION
**Pull request by chrisforbes**
Fixes #933.

### CODE REVIEW

**Comment by mark-lunarg** [comment]
LGTM.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -3167,6 +3167,14 @@ static bool validatePipelineDrawtimeState(layer_data const *my_data,
                         reinterpret_cast<uint64_t &>(pCB->activeRenderPass->renderPass), reinterpret_cast<uint64_t &>(pPipeline),
                         reinterpret_cast<const uint64_t &>(pPipeline->graphicsPipelineCI.renderPass), err_string.c_str());
         }
+
+        if (pPipeline->graphicsPipelineCI.subpass != pCB->activeSubpass) {
```
Comment (tobine) [inline layers/core_validation.cpp]: Is there explicit spec language for this requirement? The check looks good and makes sense but in ~5min I couldn't find spec language enforcing this so we should add it if it's not there.

Comment (chrisforbes) [inline layers/core_validation.cpp]: No, there isn't -- it's really vague about this. The closest you get is the first para of 7.2.

Comment (chrisforbes) [inline layers/core_validation.cpp]: See KhronosGroup/Vulkan-Docs#375

**Comment by chrisforbes** [comment]
Clean on the farm; pushed to master.

**Pull request closed without merging**